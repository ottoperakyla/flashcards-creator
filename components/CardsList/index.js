import React from 'react'
import styled from 'styled-components'
import Card from '../Card'
import Button from '../Button'

const CardsListContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: stretch;
`

const Title = styled.h2`
  text-align: center;
`

export default ({cards, setQuizRunning}) => {
  return (
    <div style={{textAlign: 'center'}}>
      <Title>Current Cards</Title>
      <CardsListContainer>
        {cards.map((props, idx) => 
          <Card key={idx} {...props} />
        )}
      </CardsListContainer>
      <Button onClick={() => setQuizRunning(true)}>Start quiz</Button>
    </div>
  )
}
