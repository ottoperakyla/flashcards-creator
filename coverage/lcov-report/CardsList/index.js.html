<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for CardsList/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">CardsList</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import styled from 'styled-components'
import Card from '../Card'
import Button from '../Button'
&nbsp;
const CardsListContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: stretch;
`
&nbsp;
const Title = styled.h2`
  text-align: center;
`
&nbsp;
export default ({cards, flipped, setFlipped, setQuizRunning, editCard, deleteCard}) =&gt; {
  return (
    &lt;div style={{textAlign: 'center'}}&gt;
      &lt;Title&gt;Current terms&lt;/Title&gt;
      {cards.length &gt; 0 ? (
      &lt;div&gt;
      &lt;CardsListContainer&gt;
        {cards.map((props, idx) =&gt; {
          const isFlipped = flipped.includes(props.term)
          return &lt;Card key={idx} idx={idx} {...props} deleteCard={deleteCard} editCard={editCard} flipped={flipped} isFlipped={isFlipped} setFlipped={setFlipped} /&gt;
        })}
      &lt;/CardsListContainer&gt;
      &lt;Button onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >setQuizRunning(true)}</span>&gt;Start quiz&lt;/Button&gt;
      &lt;/div&gt;
      ) : (
        &lt;div&gt;No terms yet. Add some with the form below!&lt;/div&gt;
      )}
    &lt;/div&gt;
  )
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Dec 22 2018 16:58:29 GMT+0200 (EET)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
