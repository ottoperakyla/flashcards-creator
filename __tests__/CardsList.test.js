import React from 'react'
import  { create } from 'react-test-renderer'
import {render, fireEvent} from 'react-testing-library'
import CardsList from '../components/CardsList'

test('snapshot (empty cards)', () => {
  const testCards = []
  const flipped = []
  const element = create(<CardsList flipped={flipped} cards={testCards} />)
  expect(element.toJSON()).toMatchSnapshot()
})

test('snapshot (with cards, one flipped)', () => {
  const testCards = [
    {term: "lollero", definition: "nope"}, 
    {term: "test2", definition: "yep"}
  ]
  const flipped = ['lollero']
  const element = create(<CardsList flipped={flipped} cards={testCards} />)
  expect(element.toJSON()).toMatchSnapshot()
})

test('snapshot (with cards, nothing flipped)', () => {
  const testCards = [
    {term: "lollero", definition: "nope"}, 
    {term: "test2", definition: "yep"}
  ]
  const flipped = []
  const element = create(<CardsList flipped={flipped} cards={testCards} />)
  expect(element.toJSON()).toMatchSnapshot()
})

// test('flips card when clicking on card', () => {
//   const testCards = [
//     {term: "lollero", definition: "nope"}, 
//     {term: "test2", definition: "yep"}
//   ]
//   const flipped = []
//   const {container} = render(<CardsList setFlipped={() => null} flipped={flipped} cards={testCards} />)
//   const firstCard = container.querySelector('.test')
//   console.log(firstCard.style)
//   fireEvent.click(firstCard)
//   console.log(firstCard.style)
// })