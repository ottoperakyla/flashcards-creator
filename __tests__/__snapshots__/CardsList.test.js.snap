// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot (empty cards) 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h2
    className="sc-gzVnrw hDwEie"
  >
    Current terms
  </h2>
  <div>
    No terms yet. Add some with the form below!
  </div>
</div>
`;

exports[`snapshot (with cards, nothing flipped) 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h2
    className="sc-gzVnrw hDwEie"
  >
    Current terms
  </h2>
  <div>
    <div
      className="sc-bZQynM hdGqyk"
    >
      <div>
        <div
          className="sc-bwzfXH jtosRe"
          onClick={[Function]}
        >
          <div
            className="test sc-htpNat dhmzWx"
            style={
              Object {
                "transform": false,
              }
            }
          >
            <div
              className="sc-bxivhb sc-ifAKCX bGQRXC"
            >
              <p>
                lollero
              </p>
            </div>
            <div
              className="sc-bxivhb sc-EHOje eqIzFr"
            >
              <p>
                nope
              </p>
            </div>
          </div>
        </div>
        <div>
          <button
            className="sc-bdVaJa iPNehR"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="sc-bdVaJa iPNehR"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </div>
      <div>
        <div
          className="sc-bwzfXH jtosRe"
          onClick={[Function]}
        >
          <div
            className="test sc-htpNat dhmzWx"
            style={
              Object {
                "transform": false,
              }
            }
          >
            <div
              className="sc-bxivhb sc-ifAKCX bGQRXC"
            >
              <p>
                test2
              </p>
            </div>
            <div
              className="sc-bxivhb sc-EHOje eqIzFr"
            >
              <p>
                yep
              </p>
            </div>
          </div>
        </div>
        <div>
          <button
            className="sc-bdVaJa iPNehR"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="sc-bdVaJa iPNehR"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </div>
    </div>
    <button
      className="sc-bdVaJa iPNehR"
      onClick={[Function]}
    >
      Start quiz
    </button>
  </div>
</div>
`;

exports[`snapshot (with cards, one flipped) 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h2
    className="sc-gzVnrw hDwEie"
  >
    Current terms
  </h2>
  <div>
    <div
      className="sc-bZQynM hdGqyk"
    >
      <div>
        <div
          className="sc-bwzfXH jtosRe"
          onClick={[Function]}
        >
          <div
            className="test sc-htpNat dhmzWx"
            style={
              Object {
                "transform": "rotateY(180deg)",
              }
            }
          >
            <div
              className="sc-bxivhb sc-ifAKCX bGQRXC"
            >
              <p>
                lollero
              </p>
            </div>
            <div
              className="sc-bxivhb sc-EHOje eqIzFr"
            >
              <p>
                nope
              </p>
            </div>
          </div>
        </div>
        <div>
          <button
            className="sc-bdVaJa iPNehR"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="sc-bdVaJa iPNehR"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </div>
      <div>
        <div
          className="sc-bwzfXH jtosRe"
          onClick={[Function]}
        >
          <div
            className="test sc-htpNat dhmzWx"
            style={
              Object {
                "transform": false,
              }
            }
          >
            <div
              className="sc-bxivhb sc-ifAKCX bGQRXC"
            >
              <p>
                test2
              </p>
            </div>
            <div
              className="sc-bxivhb sc-EHOje eqIzFr"
            >
              <p>
                yep
              </p>
            </div>
          </div>
        </div>
        <div>
          <button
            className="sc-bdVaJa iPNehR"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="sc-bdVaJa iPNehR"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </div>
    </div>
    <button
      className="sc-bdVaJa iPNehR"
      onClick={[Function]}
    >
      Start quiz
    </button>
  </div>
</div>
`;
